types/react-autosuggest/index.d.ts

```diff
@@ -128,7 +128,17 @@ declare namespace Autosuggest {
         suggestion: TSuggestion,
         params: RenderSuggestionParams,
     ) => React.ReactNode;
-    type ShouldRenderSuggestions = (value: string) => boolean;
+    type ShouldRenderSuggestions = (
+        value: string,
+        reason:
+            | 'input-changed'
+            | 'input-focused'
+            | 'input-blurred'
+            | 'escape-pressed'
+            | 'suggestions-revealed'
+            | 'suggestions-updated'
+            | 'render'
+    ) => boolean;
```

Comment:
Maybe make `Reason` a type instead of doing it in line?

Otherwise looks good 👍

Reply:
Good idea! I've extracted both this type for reasons here and for FetchRequest (overlapping, but slightly different) into standalone types.

Reply:
Nice!

Thanks for doing that